src/components/Sortable/table/index.tsx

```diff
@@ -0,0 +1,178 @@
+    const {
+        items,
+        onChange,
+        disabled,
+        getItemId = item => item?.id ?? '',
+        columns,
+    } = props;
+    const [activeItem, setActiveItem] = useState<Active | undefined>();
+    const classes = useStyles();
+    const sensors = useSensors(
+        useSensor(PointerSensor),
+        useSensor(KeyboardSensor, {
+            coordinateGetter: sortableKeyboardCoordinates,
+        }),
+    );
+
+    const ids = useMemo(() => items.map(getItemId), [items, getItemId]);
+    const itemsMap = useMemo(
+        () =>
+            items.reduce<{ [key: string]: any }>((acc, cur) => {
+                acc[getItemId(cur)] = cur;
+                return acc;
+            }, {}),
+        [items, getItemId],
+    );
+    const handleDragEnd = useCallback(
+        (event: DragEndEvent) => {
+            setActiveItem(undefined);
+            const { active, over } = event;
+            if (over && active.id !== over?.id) {
+                const oldIndex = ids.indexOf(active.id);
+                const newIndex = ids.indexOf(over.id);
+                const newIds = arrayMove(ids, oldIndex, newIndex);
+
+                onChange(newIds.map(id => itemsMap[id]));
+            }
+        },
+        [ids, itemsMap, onChange],
+    );
+    return (
+        <DndContext
+            sensors={sensors}
+            collisionDetection={closestCenter}
+            onDragEnd={handleDragEnd}
+            onDragStart={({ active }) => {
+                setActiveItem(active);
+            }}
+            modifiers={[restrictToVerticalAxis]}
+        >
+            <SortableContext
+                disabled={disabled}
+                items={ids}
+                strategy={verticalListSortingStrategy}
+            >
+                <Table size="small" className={classes.table}>
+                    <TableHead>
+                        <TableRow>
+                            <TableCell
+                                className={classNames(
+                                    classes.headerCell,
+                                    classes.sortCell,
+                                )}
+                            />
+                            {columns.map((col, ind) => (
+                                <TableCell
+                                    key={col.accessor}
+                                    className={classNames(
+                                        classes.headerCell,
+                                        ind + 1 === columns.length &&
+                                            classes.lastHeaderCell,
+                                    )}
+                                >
+                                    {col.Header}
+                                </TableCell>
+                            ))}
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {ids.map(id => (
```

Comment:
Tiens, why do you iterate on `ids` and not `items` ?

Reply:
Ok I see why. I would personally have passed the current item on each row & cell instead of the id and the map of items.

Reply:
In fact this is part of the code from your component but indeed !